### PR TITLE
Review Task Mover - Conflict Resolution options

### PR DESCRIPTION
**Pull request by davidjgonzalez**
Addresses #754 

AEM 6.2 preview packages (defaults behavior to new-version)

* [acs-aem-commons-content-3.1.1-SNAPSHOT-min.zip](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/370061/acs-aem-commons-content-3.1.1-SNAPSHOT-min.zip)
* [acs-aem-commons-content-3.1.1-SNAPSHOT.zip](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/370062/acs-aem-commons-content-3.1.1-SNAPSHOT.zip)
* [acs-aem-commons-review-task-overlay-2.0.1-AEM62.zip](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/370063/acs-aem-commons-review-task-overlay-2.0.1-AEM62.zip)
* [acs-aem-commons-review-task-osgi-config-2.0.0.zip](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/367527/acs-aem-commons-review-task-osgi-config-2.0.0.zip)

review-task-osgi
review-task

### CODE REVIEW

File: bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java [hunk 1]
```diff
@@ -199,23 +248,105 @@ public void run() {
...
+         * @param resourceResolver the ResourceResolver object
+         * @param originalAsset the asset to create a new version for
+         * @param reviewedAsset the asset to that will represent the new version
+         * @throws PersistenceException
+         */
+        private void createRevision(ResourceResolver resourceResolver, Asset originalAsset, Asset reviewedAsset) throws PersistenceException {
+            Session session = resourceResolver.adaptTo(Session.class);
+
+            // Create the new version
+            AssetManager assetManager = resourceResolver.adaptTo(AssetManager.class);
```
Comment (bstopp) [inline bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java]: Why do this when you have access to an AssetManager in the calling method?

File: bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java [hunk 2]
```diff
@@ -229,13 +360,45 @@ private void moveAsset(AssetManager assetManager, Asset asset, ValueMap taskProp
...
+                    final boolean exists = assetManager.assetExists(destAssetPath);
+
+                    if (exists) {
+                        if (StringUtils.equals(asset.getPath(), destAssetPath)) {
+                            log.info("Reviewed asset [ {} ] is already in its final location, so there is nothing to do.", asset.getPath());
+                        } else if(CONFLICT_RESOLUTION_REPLACE.equals(conflictResolution)) {
+                            assetManager.removeAsset(destAssetPath);
+                            resourceResolver.commit();
+                            assetManager.moveAsset(asset.getPath(), destAssetPath);
+                            log.info("Moved with replace [ {} ] ~> [ {} ] based on approval status [ " + status + " ]",
```
Comment (bstopp) [inline bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java]: Don't append the string, if you're using placeholders, keep using them. Same with the other log statements in this method.

File: bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java [hunk 3]
```diff
@@ -84,11 +97,18 @@
     private static final String PATH_CONTENT_DAM = "/content/dam";
```
Comment (bstopp) [inline bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java]: Use https://docs.adobe.com/docs/en/aem/6-1/ref/javadoc/com/day/cq/dam/api/DamConstants.html#MOUNTPOINT_ASSETS ?

**Comment by bstopp** [comment]
Looks good; I was afraid there was an implicit save in those JcrCopy calls, but from what i could find there isn't.

File: bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java (hunk 2 above)
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java]: IIRC the logger methods only allow 2 place holders, no? Would require a String.format(..) to use 2+  ... no?

Comment (bstopp) [inline bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java]: There is a String, Object[] which is resolved when more than 2 placeholder args are used. http://www.slf4j.org/apidocs/org/slf4j/Logger.html#info(java.lang.String, java.lang.Object...)

Different type of overhead: String concat vs Object[] creation. The ... API is cleaner IMO

File: bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java [hunk 4]
```diff
@@ -367,29 +368,29 @@ private void moveAsset(ResourceResolver resourceResolver, AssetManager assetMana
...
                             log.info("Reviewed asset [ {} ] is already in its final location, so there is nothing to do.", asset.getPath());
-                        } else if(CONFLICT_RESOLUTION_REPLACE.equals(conflictResolution)) {
+                        } else if (CONFLICT_RESOLUTION_REPLACE.equals(conflictResolution)) {
                             assetManager.removeAsset(destAssetPath);
                             resourceResolver.commit();
                             assetManager.moveAsset(asset.getPath(), destAssetPath);
-                            log.info("Moved with replace [ {} ] ~> [ {} ] based on approval status [ " + status + " ]",
-                                    asset.getPath(), destAssetPath);
-                        } else if(CONFLICT_RESOLUTION_NEW_ASSET.equals(conflictResolution)) {
+                            log.info("Moved with replace [ {} ] ~> [ {} ] based on approval status [ {} ]",
```
Comment (bstopp) [inline bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java]: Actually:
`log.info("Moved with replace [ {} ] ~> [ {} ] based on approval status [ {} ]", asset.getPath(), destAssetPath, status)`

Should work.

Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/dam/impl/ReviewTaskAssetMoverHandler.java]: Leaving as String[] to work around IntelliJ error highlight on my side

**Pull request merged**